src/styleSheetSerializer.js

```diff
@@ -96,6 +96,9 @@ const stripUnreferencedClassNames = (result, classNames) =>
     classNames
       .reduce((acc, className) => acc.replace(new RegExp(`${className}\\s?`,'g'), ''), result);
 
+const stripUnusedStyledComponentRules = result =>
+  result.replace(/[^\n\r]*\.(?:[\w\d_]+-)?sc-[\w\d]{4,8}\s*{[^}]+}[\n\r]+/, '')
```

Comment:
During testing, the styled component id's length ranged between 5 and 6 chars. To be safe I set this to between 4 and 8, but it might be good to dial this into the actual min and max length of a styled component id.

This regex matches all CSS declarations that use selectors containing styled component class names and the trailing whitespace before the next rule, or html. When this function runs all styled component class names should be replaced already, so anything else found at this point should likely be removed.

Here's a detailed explanation of the regex:
```
/[^\n\r]*\.(?:[\w\d_]+-)?sc-[\w\d]{4,8}\s*{[^}]+}[\n\r]+/
 ^      ^ ^             ^^                      ^^-----^   | grabs the trailing whitespace after the rule
 ^      ^ ^             ^^----------------------^
 ^      ^ ^-------------^   |  grabs the styled component id with sc- prefix
 ^------^    | deals with babel-styled-components prefixes
    | grabs the start of the selector
```

Reply:
Though this method works (provided the selector is always generated one one line) it felt a bit fragile, so I've pushed a different approach where I filter to roles instead.